### PR TITLE
Adds user related operations (update, delete); updates 'createRelationship' for HAL 

### PR DESCRIPTION
**Pull request by igorkravchenko**

### CODE REVIEW

File: FitpaySDK/Rest/RestClient.swift [hunk 1]
```diff
@@ -320,9 +423,9 @@ public class RestClient : NSObject
                     "creditCardId" : "\(creditCardId)",
                     "deviceId" : "\(deviceId)"
                 ]
-                let request = self._manager.request(.PUT, "\(API_BASE_URL)/users/\(userId)/relationships", parameters: parameters, encoding: .URLEncodedInURL, headers: headers)
+                let request = self._manager.request(.PUT, url + "/relationships", parameters: parameters, encoding: .URLEncodedInURL, headers: headers)
                 request.validate().responseObject(
-                dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), completionHandler:
+                queue: dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), completionHandler:
```
Comment (bpwalford) [inline FitpaySDK/Rest/RestClient.swift]: Adding the queue argument label is causing compilation errors for me. What was the reason for adding these?

File: FitpaySDK.podspec [hunk 2]
```diff
@@ -8,7 +8,14 @@ Pod::Spec.new do |s|
...
+  s.source = { :git => 'https://github.com/fitpay/fitpay-ios-sdk.git', :branch => 'external_fpcrypto' }
+  s.source_files = 'FitpaySDK/**/*.{swift,m,h}'
+  s.private_header_files = 'FitpaySDK/**/SECP256R1KeyPairContainer+Private.h'
   s.dependency 'Alamofire', '~> 3.0'
+  s.dependency 'ObjectMapper', '~> 1.2.0'
+  s.dependency 'AlamofireObjectMapper', '~> 3.0.0'
+  s.dependency 'JWTDecode', '~> 1.0.0'
+  s.dependency 'libPusher', '~> 1.6.1'
+  s.dependency 'KeychainAccess', '~> 2.3.4'
+  s.dependency 'OpenSSL-Universal', '~> 1.0'
```
Comment (bpwalford) [inline FitpaySDK.podspec]: This is still causing issues when the project is imported to another project as a cocoapod. Isn't this the source of the static/dynamic library issue on https://github.com/krzyzanowskim/OpenSSL? Or is this PR not meant to address the outstanding cocoapod issue?

File: FitpaySDK/Rest/RestClient.swift (hunk 1 above)
Comment (igorkravchenko) [inline FitpaySDK/Rest/RestClient.swift]: This is required by latest update of AlamofireObjectMapper. Please remove Podfile.lock and Pods directory and run pod install.

Comment (bpwalford) [inline FitpaySDK/Rest/RestClient.swift]: That was the problem, thanks. (I'm still getting used to swift, apple, etc.)

File: FitpaySDK.podspec (hunk 2 above)
Comment (igorkravchenko) [inline FitpaySDK.podspec]: This is progress update to resolve static vs dynamic library issue. My plan is to remove OpenSSL completely from dependencies of the SDK.

File: FitpaySDK/Extensions/ObjCCompatibilityExtensions.swift [hunk 3]
```diff
@@ -1,621 +0,0 @@
-
```
Comment (bpwalford) [inline FitpaySDK/Extensions/ObjCCompatibilityExtensions.swift]: What's the reason for deleting the these extensions?

Comment (igorkravchenko) [inline FitpaySDK/Extensions/ObjCCompatibilityExtensions.swift]: I need them removed while updating podspec for SDK. They are to be back in the upcoming commit.

**Pull request merged**